libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/ranker/SimpleWeightedSumDataLayoutStrategyScorer.java

```diff
@@ -0,0 +1,83 @@
+package com.linkedin.openhouse.datalayout.ranker;
+
+import com.linkedin.openhouse.datalayout.strategy.DataLayoutStrategy;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Optimizing across multiple objective such as (x) Maximize file count reduction (x) Minimize
+ * compute cost used for compaction by first normalizing the objectives on a common scale and then
+ * performing scalarization using a weighted sum technique.
+ */
+public class SimpleWeightedSumDataLayoutStrategyScorer implements DataLayoutStrategyScorer {
+
+  public static final int MINIMIZATION_OBJECTIVE_FACTOR = -1;
+
+  private final double costWeight;
+  private final double gainWeight;
+
+  public SimpleWeightedSumDataLayoutStrategyScorer(double gainWeight, double costWeight) {
+    this.gainWeight = gainWeight;
+    this.costWeight = costWeight;
+  }
+
+  /**
+   * Compute scores for the data layout strategies based on the input data.
+   *
+   * @param dataLayoutStrategies the data layout strategies to score
+   * @return the data layout strategies w/ scores
+   */
+  @Override
+  public List<DataLayoutStrategy> scoreDataLayoutStrategies(
+      List<DataLayoutStrategy> dataLayoutStrategies) {
+
+    List<DataLayoutStrategy> normalizedDataLayoutStrategies = new ArrayList<>();
+    double minCost = minCost(dataLayoutStrategies);
+    double maxCost = maxCost(dataLayoutStrategies);
+    double minGain = minGain(dataLayoutStrategies);
+    double maxGain = maxGain(dataLayoutStrategies);
+
+    for (DataLayoutStrategy dataLayoutStrategy : dataLayoutStrategies) {
+      double normalizedCost = minMaxNormalize(dataLayoutStrategy.getCost(), minCost, maxCost);
+      double normalizedGain = minMaxNormalize(dataLayoutStrategy.getGain(), minGain, maxGain);
+      DataLayoutStrategy normalizedDataLayoutStrategy =
+          DataLayoutStrategy.builder()
+              .config(dataLayoutStrategy.getConfig())
+              .entropy(dataLayoutStrategy.getEntropy())
+              .cost(dataLayoutStrategy.getCost())
+              .gain(dataLayoutStrategy.getGain())
```

Comment:
Can we change `DataLayoutStrategy` builder annotation to  `@Builder(toBuilder = true)`, then call `dataLayoutStrategy.toBuilder()`